quintilesims/eks-sso
# [WIP] SSO Service

This is PR is a POC of the design proposal #1.

**Current Status:**

- [x] Auth0 authentication to the front-end service
- [x] Enable adding AWS credentials
  - [ ] ~Securely store credentials in local store / cookie~
- [x] Show kubeconfig for the cluster
- [x] Bootstrap k8 user with namespace and appropriate permissions
- [ ] ~Create deployable package (most likely as a helm chart)~

As this is a POC first, the focus has been on getting all the pieces to implement the design proposal working first. As such the codebase itself lacks unit tests and requires further iteration to cleanup (both front and backends).

**To test:**

1. start backend server `./src -> go run main.go -d`
note that the cli will require auth0 and eks cluster env vars to be set. If they are not set, an error message will be shown asking them to be set.
2. start frontend server `./src/ui -> npm start`
3. follow the screen prompts
4. confirm that the kubeconfig give you access:
   - to the cluster you deployed in
   - and only to the namespace created for the user

closes: https://github.com/quintilesims/kubernetes-addons/issues/3
